grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details_test.ts

```diff
@@ -0,0 +1,203 @@
+import {async, TestBed, ComponentFixture} from '@angular/core/testing';
+import {initTestEnvironment} from '@app/testing';
+
+import {PluginsModule} from './module';
+import { OsqueryDetails } from './osquery_details';
+import { newFlowListEntry, newFlow } from '@app/lib/models/model_test_util';
+import { FlowState, FlowListEntry, FlowResultSet, FlowResultSetState } from '@app/lib/models/flow';
+import { By } from '@angular/platform-browser';
+import { DebugElement, Component } from '@angular/core';
+import { OsqueryResult, OsqueryTable } from '@app/lib/api/api_interfaces';
+
+
+initTestEnvironment();
+
+/** Helper class to build a FlowListEntry objects in a declarative manner */
+class FlowListEntryBuilder {
```

Comment:
The builder pattern is very Java-y, coming from Java's less powerful, stricter, nominal type system. We typically use TypeScript's structural typing for less verbose code. Please re-use `model_test_util.newFlowListEntry()`. Your `withResult` and `withTable` functions are very helpful though, please do keep them!

Reply:
Could we have a short VC once you are back and discuss that? I spent some time thinking about it and playing with various different options, and the builder still seems like the cleanest way to do it. The only improvement I could think of was to set `private flowListEntry: any = newFlowListEntry();`, and I'm not sure that's the cleanest way to do it because I'm throwing away type information.